### PR TITLE
[WIP] Add audit model

### PR DESCRIPTION
**Pull request by magajh**
Adding eox-audit-model

### CODE REVIEW

File: eox_tagging/settings/common.py [hunk 1]
```diff
@@ -18,16 +18,13 @@
 
 # Application definition
 
-INSTALLED_APPS = [
-    'django.contrib.auth',
-    'django.contrib.contenttypes',
-    'rest_framework',
-    'eox_tagging',
-]
+INSTALLED_APPS = []
```
Comment (mariajgrimaldi) [inline eox_tagging/settings/common.py:21]: If you remove this I don't think the migrations will run properly, please check that

File: eox_tagging/settings/common.py [hunk 2]
```diff
@@ -18,16 +18,13 @@
 
 # Application definition
 
-INSTALLED_APPS = [
-    'django.contrib.auth',
-    'django.contrib.contenttypes',
-    'rest_framework',
-    'eox_tagging',
-]
```
Comment (mariajgrimaldi) [inline eox_tagging/settings/common.py:26]: If you remove this I don't think the migrations will run properly, please check that

File: eox_tagging/settings/test.py [hunk 3]
```diff
@@ -43,18 +36,11 @@ class SettingsClass(object):
 
 USE_TZ = True
 
-ALLOWED_HOSTS = ['*']
-
 EOX_TAGGING_SKIP_VALIDATIONS = True
 EOX_TAGGING_LOAD_PERMISSIONS = False
 EOX_TAGGING_BEARER_AUTHENTICATION = 'eox_tagging.edxapp_wrappers.backends.bearer_authentication_i_v1_test'
+EOX_TAGGING_GET_ENROLLMENT_OBJECT = "eox_tagging.edxapp_wrappers.backends.enrollment_i_v1"
+EOX_TAGGING_GET_COURSE_OVERVIEW = "eox_tagging.edxapp_wrappers.backends.course_overview_i_v1"
```
Comment (mariajgrimaldi) [inline eox_tagging/settings/test.py:43]: why is this needed?

File: eox_tagging/api/v1/viewset.py [hunk 4]
```diff
@@ -95,3 +97,55 @@ def __get_user(self):
...
+        """Creates a tag.
+
+        Args:
+            request: Django request.
+
+        Returns:
+            Django Response.
+        """
+        @audit_method(action="create-tag-api-v1")
+        def create_tag(request_data):
```
Comment (felipemontoya) [inline eox_tagging/api/v1/viewset.py:111]: @andrey-canon since you know the audit-model decorator better than anyone. Is this how it is supposed to work?
Should we be creating an internal method to unpack the data and be able to audit a call to an API?

Perhaps we should decorate the call to `dispatch` instead?

Comment (andrey-canon) [reply eox_tagging/api/v1/viewset.py:111]: if you decorate the method `create `the model with store as inputs the string representation of  self, request, args y kwargs, so this is not mandatory but this help to show  as input the data of interest

Comment (felipemontoya) [reply eox_tagging/api/v1/viewset.py:111]: Yes, that is correct, but the string representation of the request is of no use. Args  and Kwargs are null as well.

Since this is a viewset we don't have access to dispatch, right?

What if we alter the audit-model function to store a custom representation of a request instead of __str__? one that is better for our case?
